src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerRegistryEndpoint.java

```diff
@@ -247,6 +288,24 @@ public KeyMaterialFactory newKeyMaterialFactory(@CheckForNull Item context, @Non
         return token.newKeyMaterialFactory(getEffectiveUrl(), workspace, launcher, env, listener, dockerExecutable);
     }
 
+    /**
+     * Makes the credentials available locally and returns {@link KeyMaterialFactory} that gives you the parameters
+     * needed to access it.
+     * @param context The build trying to access DockerHub
+     * @param workspace a workspace being used for operations ({@link WorkspaceList#tempDir} will be applied)
+     * @param dockerExecutable as in {@link DockerTool#getExecutable}, with a 1.8+ client
+     */
+    public KeyMaterialFactory newKeyMaterialFactory(@CheckForNull Run context, @Nonnull FilePath workspace, @Nonnull Launcher launcher, @Nonnull EnvVars env, @Nonnull TaskListener listener, @Nonnull String dockerExecutable) throws IOException, InterruptedException {
```

Comment:
Please deprecate the old overload so we can keep track of which one is actually called from `docker-workflow`.

(I have a mind to just move those two steps here, actually, so `docker-workflow` can be left to rot.)